### PR TITLE
Ports carrier visible huggers from CM

### PR DESCRIPTION
**Pull request by Blundir**
## About The Pull Request

Partly ports CM hugger layer for carrier, i tweaked the pallete for it to match our carrier sprites.
dreamseeker_ldbjEt4fKD

## Why It's Good For The Game

More visual depth and detail, general visual improvement that will help visually identify amount of huggers the carrier has, good for both xenos and marines.

## Changelog

:cl:
add: ported carrier hugger overlay
imageadd: added hugger images for carrier
/:cl:

### CODE REVIEW

**Comment by FosterGrime-Heymynameslouie** [comment]
can finally see death in 4k!

spededmehgd

**Comment by iLikeMemes** [comment]
Bald carrier....

**Review by Kuro020 — APPROVED** [review]

File: code/modules/mob/living/carbon/xenomorph/castes/carrier/carrier.dm [hunk 1]
```diff
@@ -25,3 +28,52 @@
...
+	for(var/i in hugger_image_index)
+		if(stat == DEAD)
+			hugger_overlays_icon.overlays += icon(icon, "clinger_[i] Knocked Down")
+		else if(lying_angle)
+			if((resting || IsSleeping()) && (!IsParalyzed() && !IsUnconscious() && health > 0))
+				hugger_overlays_icon.overlays += icon(icon, "clinger_[i] Sleeping")
+			else
+				hugger_overlays_icon.overlays +=icon(icon, "clinger_[i] Knocked Down")
+		else
+			hugger_overlays_icon.overlays +=icon(icon, "clinger_[i]")
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/castes/carrier/carrier.dm:59]: do not use icon it is slow

File: code/modules/mob/living/carbon/xenomorph/castes/carrier/carrier.dm [hunk 2]
```diff
@@ -17,6 +17,9 @@
 		/mob/living/carbon/xenomorph/proc/vent_crawl,
 	)
 
+	var/list/hugger_image_index = list()
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/castes/carrier/carrier.dm:20]: this list does not need to exist

File: code/modules/mob/living/carbon/xenomorph/castes/carrier/carrier.dm [hunk 3]
```diff
@@ -17,6 +17,9 @@
...
+	var/mutable_appearance/hugger_overlays_icon
+
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/castes/carrier/carrier.dm:22]: autodoc

File: code/modules/mob/living/carbon/xenomorph/castes/carrier/carrier.dm [hunk 4]
```diff
@@ -25,3 +28,52 @@
...
+			if((resting || IsSleeping()) && (!IsParalyzed() && !IsUnconscious() && health > 0))
+				hugger_overlays_icon.overlays += icon(icon, "clinger_[i] Sleeping")
+			else
+				hugger_overlays_icon.overlays +=icon(icon, "clinger_[i] Knocked Down")
+		else
+			hugger_overlays_icon.overlays +=icon(icon, "clinger_[i]")
+
+	overlays += hugger_overlays_icon
+
+/mob/living/carbon/xenomorph/carrier/proc/update_icon_maths(number)
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/castes/carrier/carrier.dm:63]: this is not inlined for no reason

File: code/modules/mob/living/carbon/xenomorph/castes/carrier/carrier.dm [hunk 5]
```diff
@@ -25,3 +28,52 @@
...
+			while(length(hugger_image_index) != number)
+				for(var/i in hugger_image_index)
+					if(locate(i) in funny_list)
+						funny_list -= i
+				hugger_image_index += funny_list[rand(1,length(funny_list))]
+
+/mob/living/carbon/xenomorph/carrier/Initialize(mapload, h_number)
+	. = ..()
+
+	hugger_overlays_icon = mutable_appearance('icons/Xeno/2x2_Xenos.dmi',"empty")
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/castes/carrier/carrier.dm:79]: put this just below the mob definition not at the very bottom

Comment (TiviPlus) [reply code/modules/mob/living/carbon/xenomorph/castes/carrier/carrier.dm:79]: also
```suggestion
/mob/living/carbon/xenomorph/carrier/Initialize(mapload, h_number)
	. = ..()
	hugger_overlays_icon = mutable_appearance('icons/Xeno/2x2_Xenos.dmi',"empty")
```

File: code/modules/mob/living/carbon/xenomorph/castes/carrier/carrier.dm [hunk 6]
```diff
@@ -25,3 +28,52 @@
...
+	if(length(hugger_image_index) != number)
+		if(length(hugger_image_index) > number)
+			while(length(hugger_image_index) != number)
+				hugger_image_index -= hugger_image_index[length(hugger_image_index)]
+		else
+			while(length(hugger_image_index) != number)
+				for(var/i in hugger_image_index)
+					if(locate(i) in funny_list)
+						funny_list -= i
+				hugger_image_index += funny_list[rand(1,length(funny_list))]
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/castes/carrier/carrier.dm:74]: this proc is overcomplicated and makes no sense to exist when you can just iterate in a for loop for mutable appearance creation

**Review by TiviPlus — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/xenomorph/castes/carrier/carrier.dm (hunk 6 above)
Comment (Blundir) [reply code/modules/mob/living/carbon/xenomorph/castes/carrier/carrier.dm:74]: yeah, i just generally ported cm code directly, gonna look into it later

**Comment by Blundir** [comment]
gonna close for now

**Pull request closed without merging**